airbyte-integrations/bigquery-destination/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -0,0 +1,329 @@
+
+  @Override
+  public DestinationConnectionSpecification spec() throws IOException {
+    // return a jsonschema representation of the spec for the integration.
+    final String resourceString = MoreResources.readResource("spec.json");
+    return Jsons.deserialize(resourceString, DestinationConnectionSpecification.class);
+  }
+
+  @Override
+  public StandardCheckConnectionOutput check(JsonNode config) {
+    try {
+      String datasetId = config.get("dataset_id").asText();
+
+      QueryJobConfiguration queryConfig = QueryJobConfiguration
+          .newBuilder(String.format("SELECT * FROM %s.INFORMATION_SCHEMA.TABLES LIMIT 1;", datasetId))
+          .setUseLegacySql(false)
+          .build();
+
+      final ImmutablePair<Job, String> result = executeQuery(getBigQuery(config), queryConfig);
+      if (result.getLeft() != null) {
+        return new StandardCheckConnectionOutput().withStatus(Status.SUCCESS);
+      } else {
+        return new StandardCheckConnectionOutput().withStatus(Status.FAILURE).withMessage(result.getRight());
+      }
+    } catch (Exception e) {
+      return new StandardCheckConnectionOutput().withStatus(Status.FAILURE).withMessage(e.getMessage());
+    }
+  }
+
+  private BigQuery getBigQuery(JsonNode config) {
+    String projectId = config.get("project_id").asText();
+    String credentialsString = config.get("credentials_json").asText();
+    try {
+      ServiceAccountCredentials credentials = ServiceAccountCredentials.fromStream(new ByteArrayInputStream(credentialsString.getBytes()));
+
+      return BigQueryOptions.newBuilder()
+          .setProjectId(projectId)
+          .setCredentials(credentials)
+          .build()
+          .getService();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static ImmutablePair<Job, String> executeQuery(BigQuery bigquery, QueryJobConfiguration queryConfig) {
+    JobId jobId = JobId.of(UUID.randomUUID().toString());
+    Job queryJob = bigquery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
+    return executeQuery(queryJob);
+  }
+
+  private static ImmutablePair<Job, String> executeQuery(Job queryJob) {
+    final Job completedJob = waitForQuery(queryJob);
+    if (completedJob == null) {
+      throw new RuntimeException("Job no longer exists");
+    } else if (completedJob.getStatus().getError() != null) {
+      // You can also look at queryJob.getStatus().getExecutionErrors() for all
+      // errors, not just the latest one.
+      return ImmutablePair.of(null, (completedJob.getStatus().getError().toString()));
+    }
+
+    return ImmutablePair.of(completedJob, null);
+  }
+
+  private static Job waitForQuery(Job queryJob) {
+    try {
+      return queryJob.waitFor();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public StandardDiscoverSchemaOutput discover(JsonNode config) {
+    throw new RuntimeException("Not Implemented");
```

Comment:
what should we do here?

Reply:
I'll just remove it from the iface for now. will do in a later PR.